tests/integration/devfile/cmd_devfile_catalog_test.go

```diff
@@ -140,4 +140,122 @@ var _ = Describe("odo devfile catalog command tests", func() {
 		})
 	})
 
+	Context("odo devfile catalog command tests without kubeconfig", func() {
+		JustBeforeEach(func() {
+			helper.LocalKubeconfigSet("")
+		})
+
+		It("should list all supported devfile components", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components")
+			wantOutput := []string{
+				"Odo Devfile Components",
+				"NAME",
+				"java-springboot",
+				"java-openliberty",
+				"java-quarkus",
+				"DESCRIPTION",
+				"REGISTRY",
+				"DefaultDevfileRegistry",
+			}
+			helper.MatchAllInOutput(output, wantOutput)
+		})
+
+		It("should list devfile components in json format with -o flag", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components", "-o", "json")
+			wantOutput := []string{
+				"odo.dev/v1alpha1",
+				"devfileItems",
+				"java-openliberty",
+				"java-springboot",
+				"nodejs",
+				"java-quarkus",
+				"java-maven",
+			}
+			helper.MatchAllInOutput(output, wantOutput)
+		})
+	})
+
+	Context("When executing catalog list components with registry that is not set up properly, without kubeconfig set", func() {
+		JustBeforeEach(func() {
+			helper.LocalKubeconfigSet("")
+		})
+
+		It("should list components from valid registry", func() {
+			helper.CmdShouldPass("odo", "registry", "add", "fake", "http://fake")
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components")
+			helper.MatchAllInOutput(output, []string{
+				"Odo Devfile Components",
+				"java-springboot",
+				"java-quarkus",
+			})
+			helper.CmdShouldPass("odo", "registry", "delete", "fake", "-f")
+		})
+	})
+
+	Context("When executing catalog describe component with a component name with a single project, without kubeconfig set", func() {
+		JustBeforeEach(func() {
+			helper.LocalKubeconfigSet("")
+		})
+
+		It("should only give information about one project", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "describe", "component", "java-openliberty")
+			Expect(output).To(MatchRegexp("origin: .+"))
+		})
+	})
+
+	Context("When executing catalog describe component with a component name with no starter projects, without kubeconfig set", func() {
+		JustBeforeEach(func() {
+			helper.LocalKubeconfigSet("")
+		})
+
+		It("should print message that the component has no starter projects", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "describe", "component", "java-maven")
+			helper.MatchAllInOutput(output, []string{"The Odo devfile component \"java-maven\" has no starter projects."})
+		})
+	})
+
+	Context("When executing catalog describe component with a component name with multiple components, without kubeconfig set", func() {
```

Comment:
@dev-gaur It's really nice that we are covering all possible scenarios for catalog describe. However one thing that I think can't we cover possible scenarios in unit test. I am saying this because it seems like a single line conditional scenarios.